spacy_transformers/align.py

```diff
@@ -135,14 +135,24 @@ def get_alignment(spans: List[Span], wordpieces: List[List[str]]) -> Ragged:
     """
     if len(spans) != len(wordpieces):
         raise ValueError("Cannot align batches of different sizes.")
+    if special_tokens is None:
+        special_tokens = []
     # Tokens can occur more than once, and we need the alignment of each token
     # to its place in the concatenated wordpieces array.
     token_positions = get_token_positions(spans)
     alignment: List[Set[int]] = [set() for _ in range(len(token_positions))]
     wp_start = 0
     for i, (span, wp_toks) in enumerate(zip(spans, wordpieces)):
         sp_toks = [token.text for token in span]
-        span2wp, wp2span = get_alignments(sp_toks, wp_toks)
+        wp_toks_filtered = wp_toks
+        # In the case that the special tokens do not appear in the text, filter
+        # them out for alignment purposes so that special tokens like "<s>" are
+        # not aligned to the character "s" in the text. (If the special tokens
+        # appear in the text, it's not possible to distinguish them from the
+        # added special tokens, so they may be aligned incorrectly.)
+        if not any([special in span.text for special in special_tokens]):
+            wp_toks_filtered = [tok if tok not in special_tokens else "" for tok in wp_toks]
+        span2wp, wp2span = get_alignments(sp_toks, wp_toks_filtered)
```

Comment:
Okay so `wp_toks_filtered` uses the placeholder `""` to make sure that the indices aren't off, right? I guess that works, but it makes me nervous.

Would it be cleaner to provide `get_alignments` the indices that are special tokens? If so, let's add it as a TODO for the `spacy-tokenizations` library. I expect we won't want to do it until we do the Cython translation.

Reply:
Yes, this is to preserve the indices.

I'd rather not modify the alignment library if we don't have to, or I'd rather do it with a util in `spacy-transformers` than in `spacy-alignments`. `spacy-alignments` is a really really thin wrapper with no modified code aside from the setup at this point and it'll be easier to maintain if it stays that way.

Reply:
As a note, it would be potentially useful to provide a mask for the alignment for truncated tokens, too, but it currently seems too difficult to track truncated tokens in the pipeline.